### PR TITLE
api: add support for topology based provisioning

### PR DESCRIPTION
**Pull request by Madhu-1**
This PR exposes new fields in the nodeplugin spec where the user can specify the domain labels for the cephcsi to advertise its topology domain

fixes: #80

### CODE REVIEW

File: api/v1alpha1/driver_types.go [hunk 1]
```diff
@@ -166,6 +172,9 @@ type NodePluginSpec struct {
 	// Control the host mount of /etc/selinux for csi plugin pods. Defaults to false
 	//+kubebuilder:validation:Optional
 	EnableSeLinuxHostMount *bool `json:"EnableSeLinuxHostMount,omitempty"`
+	// Topology settings for the plugin pods
+	//+kubebuilder:validation:Optional
+	Topology *TopologySpec `json:"topology,omitempty"`
```
Comment (nb-ohad) [inline api/v1alpha1/driver_types.go:177]: Is topology only important for NodePlugin? what about the ControllerPlugin? shouldn't er have that in the PodCommonSpec?

File: internal/utils/csi.go [hunk 2]
```diff
@@ -393,3 +392,11 @@ func FuseMountOptionsContainerArg(options map[string]string) string {
 		"",
 	)
 }
+
+func TopologyDomainLaels(options []string) string {
```
Comment (nb-ohad) [inline internal/utils/csi.go:396]: 1. a misspelling of the word "labels"
2. We should align on the naming convention for dynamic container args (`<arg name>+ContainerArg`) 
```suggestion
func DomainLabelsContainerArg(options []string) string {
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: api/v1alpha1/driver_types.go (hunk 1 above)
Comment (Madhu-1) [reply api/v1alpha1/driver_types.go:177]: This is only required for node plugin they need to advertise topology on the node where it is running

File: internal/utils/csi.go [hunk 3]
```diff
@@ -393,3 +392,11 @@ func FuseMountOptionsContainerArg(options map[string]string) string {
 		"",
 	)
 }
+
```
Comment (nb-ohad) [inline internal/utils/csi.go:395]:
```suggestion
```

File: api/v1alpha1/driver_types.go (hunk 1 above)
Comment (nb-ohad) [reply api/v1alpha1/driver_types.go:177]: ack

**Review by nb-ohad — APPROVED** [review]

**Pull request merged**